Playground repositories overview (#1270)
WordPress/wordpress-playground: packages/docs/site/docs/01-start-here/03-repos-overview.md

> # Understanding the Playground Repositories
> There are a couple of different repos that serve different purposes in the WordPress Playground project. Here's a quick overview of each:
> 

Author: ```suggestion
Search for issues and discussions in the relevant repository:
```

## [WordPress Playground](https://github.com/WordPress/wordpress-playground/)

> This is the main repository for the WordPress Playground project. It contains the core codebase for the project, project boards and issues, and discussions around ongoing project improvements and maintenance.

Author: OK with this, edited for accuracy

> This is the main repository for the WordPress Playground project. It contains the core codebase for the project, project boards and issues, and discussions around ongoing project improvements and maintenance.

Author: ```suggestion
This is the main repository for the WordPress Playground. It contains the core codebase, issues and discussions about the project. 
```

> The repo is there to guide the core contributors to the Playground identify issues, feature requests, and other improvements that need to be made to the project.

Author: ```suggestion
```

remove / extra

> Opening issues in the WordPress Playground repository for the "online" version of the Playground, IE: playground.wordpress.net, is a great way to get visibility on a problem or feature request. But there may be a better suited repo below to open the issue, and we'll discuss that shortly.

Author: ```suggestion
Issues in the WordPress Playground repository relate to the "online" versions of the Playground. The Blueprint Builder and other public facing Playground apps should be discussed and reported here.
```

> Searching for issues is a great way to find a fix or workaround, or to see if a feature request has already been made. If you find an issue that matches your problem, you can add a comment to the issue to provide more information or to show that you are also experiencing the same problem.

Author: more straightforward

> Searching for issues is a great way to find a fix or workaround, or to see if a feature request has already been made. If you find an issue that matches your problem, you can add a comment to the issue to provide more information or to show that you are also experiencing the same problem.

Author: ```suggestion
Over time many issues will be discussed in the primary repository. If you are new to using Playground and trying it online, try searching for any issue you may be having in the main repo. 
```

Explore the WordPress Playground [repository](https://github.com/WordPress/wordpress-playground/)

---

## [Playground Tools]((https://github.com/WordPress/playground-tools/))

> This repository contains a collection of tools and utilities that are used to manage and maintain the WordPress Playground project. The tools include scripts for building and deploying the project, as well as utilities for managing the documentation and other project assets.

Author: ```suggestion
Tools and utilities for the Playground.
```

There are also advanced administration features, integrations such as the Playground Plugin, Blueprint Builder, and Interactive Code Block that live in the Playground Tools repository.

This repository is also the best place to search for issues related to `wp-now` or the `VS Code Extension` for WordPress Playground. If the request you are looking for is not specific to the online version of the Playground, it may be in the Playground Tools repository.

#### Tools that exist in the Playground Tools repository:

> - [Blueprint Builder](https://playground.wordpress.net/builder/builder.html) - a tool to create a view Blueprints side-by-side

Reviewer: Nitpick: This one is actually in the main Playground repo 😅 I'd love to move it over, just didn't get to it yet

Reviewer: But the playground-tools repo is still cool for opening issues

Author: So technically the Blueprint Builder is an online interface. The way I've been leaning towards the WordPress Playground repo was for "Playground online" - AKA playground.wordpress.net

Feel free to re-word or I can take a look at it later or tomorrow. Gonna switch gears for a little bit, thanks for checking in today!

Author: the Blueprint Builder is in the Playground repo... it's an online tool. That makes sense.

Will update...

- [Playground Block](https://wordpress.org/plugins/interactive-code-block/) - embed Playground via a block on your WordPress site
- [Interactive Code Block](https://wordpress.org/plugins/interactive-code-block/) - an interactive code editor to demonstrate and teach your readers how WordPress plugins are built.
- [Playground Plugin](https://wordpress.org/plugins/playground/) - Copy your site or test plugins easily
- [VS Code Extension](https://marketplace.visualstudio.com/items?itemName=WordPressPlayground.wordpress-playground) - Easily spin up WordPress instances in VS Code
- [`wp-now` ](https://www.npmjs.com/package/%40wp-now/wp-now)- a fully featured, extensible environment for WordPress development

> - [Pull Request Previewer](https://playground.wordpress.net/wordpress.html) - a tool to preview pull requests in the Playground

Reviewer: Nitpick: This one is actually in the main Playground repo 😅 I'd love to move it over, just didn't get to it yet

Author: It probably is helpful to actually describe what tools are in where, so devs don't have to go looking around everywhere. Copilot wrote most of the tools stuff and I filled in the details, didn't actually go digging in the repos 😆

Author: PR Previewer is in the Playground repo

Will update

Reviewer: > Copilot wrote most of the tools stuff and I filled in the details,

It kinda has that AI vibe, e.g. it discusses generic terms like issues and pull requests. I suggest not relying on copilot too much – I know I could never make it produce specific, helpful, high quality writing. Rewriting these pages by hand to share your insights with the reader would reduce the generic feel, make the content clear, and more useful.

Author: Good call. It's good for outlines and such, but it important to put the personal love into it.

- etc.

The list of Playground Tools continues to grow by the day, and we cannot wait to see what exciting tools are added next. Want to contribute? Swing on by the Playground Tools [repository](https://github.com/WordPress/playground-tools/) and see what you can do to help.

---

## [Blueprints Library](https://github.com/WordPress/blueprints-library)

> The Blueprints Library is a collection of pre-configured WordPress setups that can be used to quickly create a new WordPress site in the Playground. Each blueprint contains a set of instructions for setting up a specific type of WordPress site, along with the necessary configuration files.

Reviewer: Ah this one is a library in a "PHP library" sense, as in a codebase that will eventually run the Blueprints. Perhaps we could come with a better name. The Blueprints gallery/community space is at https://github.com/adamziel/blueprints and will be moved over under the WordPress org soon

Author: > The Blueprints Library is a collection of pre-configured utilities that can be leveraged to quickly create features for a WordPress installation, based on a series of instructions and necessary configuration files.

How about that?

--- 

I envision the Blueprints Library as an interface for parsing blueprints. Which, technically a library has a numbering system, a librarian, programming/etc - so maybe it is a good term. Part of the Blueprints Library is the Blueprints API and underlying programming mechanisms that make it work.

Reviewer: I meant that repository is a software library as in https://en.wikipedia.org/wiki/Library_(computing), not as a place where Blueprints are filed and managed.

Author: Vastly improved
